cloudify_agent/installer/config/configuration.py

```diff
@@ -162,6 +166,51 @@ def cfy_agent_attributes(cloudify_agent):
         cloudify_agent['manager_ip'] = get_manager_ip()
 
 
+def directory_attributes(cloudify_agent):
+    if not cloudify_agent.get('agent_dir'):
+        name = cloudify_agent['name']
+        basedir = cloudify_agent['basedir']
+        if cloudify_agent['windows']:
+            agent_dir = '{0}\\{1}'.format(basedir, name)
+        else:
+            agent_dir = os.path.join(basedir, name)
+        cloudify_agent['agent_dir'] = agent_dir
+
+    if not cloudify_agent.get('workdir'):
+        agent_dir = cloudify_agent['agent_dir']
+        if cloudify_agent['windows']:
+            workdir = '{0}\\{1}'.format(agent_dir, 'work')
+        else:
+            workdir = os.path.join(agent_dir, 'work')
+        cloudify_agent['workdir'] = workdir
+
+    if not cloudify_agent.get('envdir'):
+        agent_dir = cloudify_agent['agent_dir']
+        if cloudify_agent['windows']:
+            envdir = '{0}\\{1}'.format(agent_dir, 'env')
+        else:
+            envdir = os.path.join(agent_dir, 'env')
+        cloudify_agent['envdir'] = envdir
+
+
+@group('installation')
+def _add_installation_defaults(cloudify_agent):
+    pass
+
+
+@group('cfy-agent')
+def _add_cfy_agent_defaults(cloudify_agent):
+    pass
+
+
+def reinstallation_attributes(cloudify_agent):
+    _cfy_agent_attributes_no_defaults(cloudify_agent)
+    _add_cfy_agent_defaults(cloudify_agent)
+    if cloudify_agent.get('basedir'):
```

Comment:
There are two options:
1. We are installing new agent using old agent in version 3.3 - then we use basedir from cloudify_agent struct of an old agent.
2. We are installing new agent using old agent in version pre 3.3 - then we do not have basedir here. It is detected in https://github.com/codilime/cloudify-agent/blob/CFY-3040_agents_update/cloudify_agent/shell/commands/installer.py#L41
After that, we add directory_attributes.